CommandLine/emake/EnigmaCallbacks.cpp

```diff
@@ -11,24 +11,24 @@ void* CallBack::OutputThread(void*)
 {
   if (_outFile)
   {
-    _outFile.seekg (0, _outFile.end);
-
-    int length = _outFile.tellg();
-
-    _outFile.seekg (0, _outFile.beg);
-
-    char * buffer = new char [length];
-
-    _outFile.read (buffer,length);
-
-    std::cout << buffer;
-
-    _outFile.close();
-
-    delete[] buffer;
+    int pos = 0;
+    while (!_outFile.eof()) {
+      _outFile.seekg(0, _outFile.end);
+      int length = (int)_outFile.tellg() - pos;
+      _outFile.seekg(pos, _outFile.beg);
+ 
+      if (length > 0) {
+		char* buffer = new char[length + 1];
+		_outFile.read(buffer, length);
+		buffer[length] = '\0';
+		pos += length;
+		std::cout << buffer;
+		delete[] buffer;
+      }
+    }
```

Comment:
@JoshDreamland does this code I wrote for him look correct? It reads from the output file in a thread and prints it to the screen.